spec.bs

```diff
@@ -4249,13 +4292,36 @@ The <dfn for=Navigator method>updateAdInterestGroups()</dfn> method steps are:
         is not null;
       * |ig|'s [=interest group/estimated size=] is greater than 50 KB.
     1. Set |ig|'s [=interest group/next update after=] to the [=current wall time=] plus 24 hours.
+    1. Set |ig|'s [=interest group/last updated=] to the [=current wall time=].
     1. [=list/Replace=] the [=interest group=] that has |ig|'s [=interest group/owner=] and
       [=interest group/name=] in the [=user agent=]'s [=interest group set=] with |ig|.
     1. <i id=abort-update>Abort update</i>: We jump here if some part of the
       [=interest group=] update failed. [=iteration/Continue=] to the next [=interest group=] update.
 
 </div>
 
+<div algorithm>
+
+To <dfn>process updateIfOlderThanMs</dfn> given an [=origin=] |buyer|, and an [=ordered map=]
+|perInterestGroupData| whose [=map/keys=] are [=interest group/name=] [=strings=] and whose
+[=map/values=] are [=bidding signals per interest group data=]:
+
+1. [=map/For each=] |igName| → |perIgData| of |perInterestGroupData|, the following steps should be
+  done atomically:

```

Comment:
Could you add a note below this algorithm just indicating why "atomically" is load-bearing? Just something saying that these are DB operations that might not run atomically with respect to the database, but need to in this algorithm, and as such the atomicity is explicitly specified here, unlike other algorithms.

Reply:
The more I think about it, these operations don't need to be atomic -- it's OK if some time passes between the read (which establishes eligibility, which will never go from eligible to non-eligible since we're checking "older than") and the write (which allows the IG to participate in the next update operation, which probably will be the post-auction update), with other writes happening in between, since the net effect (the IG is allowed to update) is the same (unless the clock decreases after the write -- the implementation also uses a non-monotonic clock to handle valid values across restarts -- that's why this update is considered best-effort).

So, I just removed the word "atomically" altogether.